### ISSUE TITLE
Tech skill tree

### ISSUE THREAD

**Issue by jonathf** [expertanalytics/cvcreator#50]
This builds on the ideas of #35, tries to solve #21, and pave the way for #20 and #22.

I propose we create a tech skill tree E.g.
```
* Platforms
   * OS
      Apple
      Linux
      Microsoft Windows
   * Cloud Computing
      Docker
      Kubernetes
* Languages
   * Scripting
      Bash
      Python
   * Performance code
      C
      Rust
   C++
* Tools
...
```
Think of it as a nested fold structure with files.
E.g. `Languages/Scripting/Bash` can be grouped with both `Languages` and `Scripting`.

This allows us some wiggle room to specificity in the CV, as there might be many employees that have too few elements in a specific sub-category for it to make sense to have them clustered in only one specific way. But if they show the skills for one level up, the list gets wider. This is also an advantage from an aggregation point of view, as it allows different grain sized filters.

If this sounds like a good idea I can make a PR with a first pass of such a nested list containing all our skills we share in the XAL CVs currently that we can discuss the particulars on.

**Comment by mandus** [comment]
@vgeck also made some attempts of aggregating info - so maybe he should be assigned on this one too?

**Comment by jonathf** [comment]
Yeah, I'd love to hear your input, @vgeck.

**Comment by jonathf** [comment]
Basic tree with fewer skills have now been merged into v1.0.
